v2/datastream-to-spanner/terraform/samples/postgresql-end-to-end/main.tf

```diff
@@ -0,0 +1,284 @@
+}
+
+# Pub/Sub Notification on GCS Bucket
+resource "google_storage_notification" "bucket_notification" {
+  depends_on = [
+    google_project_service.enabled_apis,
+    google_pubsub_topic_iam_member.gcs_publisher_role
+  ] # Create a bucket notification using the created pubsub topic.
+  bucket             = google_storage_bucket.datastream_bucket.name
+  object_name_prefix = var.datastream_params.stream_prefix_path
+  payload_format     = "JSON_API_V1"
+  topic              = google_pubsub_topic.datastream_topic.id
+  event_types        = ["OBJECT_FINALIZE"]
+}
+
+# Pub/Sub Subscription for the created notification
+resource "google_pubsub_subscription" "datastream_subscription" {
+  depends_on = [
+    google_project_service.enabled_apis,
+    google_storage_notification.bucket_notification
+  ] # Create the subscription once the notification is created.
+  name  = "${google_pubsub_topic.datastream_topic.name}-sub"
+  topic = google_pubsub_topic.datastream_topic.id
+  labels = {
+    "migration_id" = local.migration_id
+  }
+}
+
+# GCS Target Connection Profile
+resource "google_datastream_connection_profile" "target_gcs" {
+  depends_on = [
+    google_project_service.enabled_apis,
+    google_storage_notification.bucket_notification
+  ] # Create the target profile once the bucket and its notification is created.
+  display_name          = "${local.migration_id}-${var.datastream_params.target_connection_profile_id}"
+  location              = var.common_params.region
+  connection_profile_id = "${local.migration_id}-${var.datastream_params.target_connection_profile_id}"
+
+  gcs_profile {
+    bucket    = google_storage_bucket.datastream_bucket.name
+    root_path = var.datastream_params.gcs_root_path
+  }
+  labels = {
+    "migration_id" = local.migration_id
+  }
+}
+
+# Datastream Stream (PostgreSQL to GCS)
+resource "google_datastream_stream" "postgresql_to_gcs" {
+  depends_on = [
+    google_project_service.enabled_apis,
+    google_pubsub_subscription.datastream_subscription
+  ]
+  # Create the stream once the source and target profiles are created along with the subscription.
+  stream_id     = "${local.migration_id}-${var.datastream_params.stream_id}"
+  location      = var.common_params.region
+  display_name  = "${local.migration_id}-${var.datastream_params.stream_id}"
+  desired_state = "RUNNING"
+  backfill_all {
+  }
+
+  source_config {
+    source_connection_profile = google_datastream_connection_profile.source_postgresql.id
+
+    postgresql_source_config {
+      publication      = var.datastream_params.postgresql_publication
+      replication_slot = var.datastream_params.postgresql_replication_slot
+      # max_concurrent_cdc_tasks      = var.datastream_params.max_concurrent_cdc_tasks
+      max_concurrent_backfill_tasks = var.datastream_params.max_concurrent_backfill_tasks
+      include_objects {
+        postgresql_schemas {
+          schema = "public"
```

Comment:
Why only the `public` schema?
User should be given the option to specify the schemas they want to migrate, right?

Reply:
Done